server/handlers/logic.go

```diff
@@ -0,0 +1,89 @@
+package handlers
```

Comment:
maybe `roles.go`?  I'm not sold on that name either but seems a little more descriptive